curvefs/src/metaserver/inode_storage.cpp

```diff
@@ -572,5 +573,22 @@ MetaStatusCode InodeStorage::GetVolumeExtentByOffset(uint32_t fsId,
     return MetaStatusCode::STORAGE_INTERNAL_ERROR;
 }
 
+MetaStatusCode
+InodeStorage::AddS3ChunkInfoList(uint32_t fsId, uint64_t inodeId,
```

Comment:
maybe you can use the exist function `AddS3ChunkInfoList` and distinguished by param, if `txn==null`, we use `kvStorage_->SSet()`, else use `txn->SSet()`.